lib/thrift/parser/resolver.ex

```diff
@@ -30,6 +30,16 @@ defmodule Thrift.Parser.Resolver do
       end
     end)
 
-    Map.merge(state, new_mappings)
+    existing_keys = Map.keys(state) |> MapSet.new
+    new_keys = Map.keys(new_mappings) |> MapSet.new
+    shared_keys = MapSet.intersection(existing_keys, new_keys) |> MapSet.to_list
```

Comment:
Yeah will do. That will be more clear.

For context, I often write similar code because it's O(1) to test whether a list has zero or 1+ elements, but it's O(n) to get its exact size. But this is not performance sensitive code, and anyway it's usually zero, so let's favor clarity.